state/upgrades.go

```diff
@@ -815,3 +815,64 @@ func addStorageInstanceConstraints(st *State) error {
 	}
 	return nil
 }
+
+var splitBatchRemoveSize = 1000
+
+// SplitLogCollections moves log entries from the old single log collection
+// to the log collection per environment.
+func SplitLogCollections(st *State) error {
+	session := st.MongoSession()
+	db := session.DB(logsDB)
+	oldLogs := db.C("logs")
```

Comment:
It turned out @howbazaar already had handling for the remove failing if the collection didn't exist - calling `db.C("logs")` doesn't create the collection, it is created lazily on insert. I added a test for that case and verified that it failed if the dropping-nonexistent-collection error handling wasn't there.